### DISCUSSION TITLE
LSP4E 0.26.5 Released

### DISCUSSION THREAD

**Discussion by rubenporras** [eclipse/lsp4e#1102] [Announcements]
## New features

* Add diagnostic code as suffix to marker message by @sebthom in https://github.com/eclipse/lsp4e/pull/1065
* Make diagnostic marker messages configurable by @sebthom in https://github.com/eclipse/lsp4e/pull/1066
* Add hide fields toolbar button to outline view (fix #1073) by @travkin79 in https://github.com/eclipse/lsp4e/pull/1075
* Allow access to CodeAction in CodeActionMarkerResolution. by @andrewL-avlq in https://github.com/eclipse/lsp4e/pull/1087
* Introduce LanguageServerWrapper#getServerCapabilitiesAsync by @sebthom in https://github.com/eclipse/lsp4e/pull/1098

## Performance improvementes

* Improve StreamProxyInputStream by @sebthom in https://github.com/eclipse/lsp4e/pull/1068
* Improve DSPLaunchDelegate debug adapter monitor performance by @sebthom in https://github.com/eclipse/lsp4e/pull/1080

## Fixes

* DocumentInputStream does not handle surrogate pairs correctly by @sebthom in https://github.com/eclipse/lsp4e/pull/1069
* Memory leak in EditorToOutlineAdapterFactory by @sebthom in https://github.com/eclipse/lsp4e/pull/1081
* Potential race condition in LSPProgressManager#report by @sebthom in https://github.com/eclipse/lsp4e/pull/1085
* Don't log same BadLocationException excessively by @sebthom in https://github.com/eclipse/lsp4e/pull/1083
* Respect inlayhint padding by @BoykoAlex in https://github.com/eclipse/lsp4e/pull/1089
* Fix disappearing sort button in outline view (fix #1091) by @travkin79 in https://github.com/eclipse/lsp4e/pull/1092
* Fix NPE seen in GitHub Actions runs by @rubenporras in https://github.com/eclipse/lsp4e/pull/1096
* Adapt resolveCodeAction to handle timeout and interrupted exception by @rubenporras in https://github.com/eclipse/lsp4e/pull/1099
* Propagate LanguagServerWrapper startup exception correctly by @ava-fred in https://github.com/eclipse/lsp4e/pull/1044

## Code improvements

* Enable NonNullByDefault for lsp4e.debug and lsp4e.jdt plugins by @sebthom in https://github.com/eclipse/lsp4e/pull/1060
* Make tm4e an optional dependency by @sebthom in https://github.com/eclipse/lsp4e/pull/1064
* Resolve sonar findings by @sebthom in https://github.com/eclipse/lsp4e/pull/1067
* Remove duplicate code from lsp4e.debug by @sebthom in https://github.com/eclipse/lsp4e/pull/1071
* Use `var` keyword and `List.of` and `Set.of` where applicable by @sebthom in https://github.com/eclipse/lsp4e/pull/1072
* Use LS.withCapability() instead of LS.withFilter() if applicable by @sebthom in https://github.com/eclipse/lsp4e/pull/1077
* Cleanup SupportedFeatures class by @sebthom in https://github.com/eclipse/lsp4e/pull/1076
* Improve handling of arrays by @sebthom in https://github.com/eclipse/lsp4e/pull/1079
* Remove duplicate null check by @rubenporras in https://github.com/eclipse/lsp4e/pull/1084
* Improve code reuse by @rubenporras in https://github.com/eclipse/lsp4e/pull/1082
* Remove unnecessary launcherFuture.isCancelled() check by @sebthom in https://github.com/eclipse/lsp4e/pull/1090
* Improve null analysis by @sebthom in https://github.com/eclipse/lsp4e/pull/1095

## Build improvements

* Mask manually entered Dash Token by @sebthom in https://github.com/eclipse/lsp4e/pull/1097

## Upgraded libraries

* Update Highlight.js to 11.10.0 by @vrubezhny in https://github.com/eclipse/lsp4e/pull/1047

**Full Changelog**: https://github.com/eclipse/lsp4e/compare/0.26.4...0.26.5

<hr /><em>This discussion was created from the release <a href='https://github.com/eclipse/lsp4e/releases/tag/0.26.5'>0.26.5</a>.</em>

**Comment by sebthom** [comment]
@rubenporras  Just a quick question about release numbers. Wouldn't new features warrant a bump of the minor version?

**Reply by rubenporras** [reply]
Since we intentionally do not use the three numbers of the release version (we stay at version 0, as far as I recall @mickaelistria wanted to stay so to indicate that the API is not fully stable), I am following the semantic versioning with some liberty, using the schema 0.MAJOR.MINOR. That is why I bumped the last number.

I would be okay to release 1.0.0 and use normal semantic versioning in the future.

**Reply by sebthom** [reply]
I don't mind staying on 0.x for now. Thanks for clarification.
